src/System.Diagnostics.Tracing/ref/System.Diagnostics.Tracing.cs

```diff
@@ -53,11 +53,36 @@ public partial class EventCommandEventArgs : System.EventArgs
         public bool DisableEvent(int eventId) { throw null; }
         public bool EnableEvent(int eventId) { throw null; }
     }
-    public partial class EventCounter : System.IDisposable
+    public abstract partial class DiagnosticCounter : System.IDisposable
     {
-        public EventCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) { }
+        public DiagnosticCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) { }
+        public void AddMetadata(string key, string value) { }
         public void Dispose() { }
+        public string DisplayName { get { throw null; } set { } }
+        public string Name { get { throw null; } }
+        public System.Diagnostics.Tracing.EventSource EventSource { get { throw null; } }
+    }
+
+    public partial class EventCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public EventCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) : base(name, eventSource) { }
         public void WriteMetric(float value) { }
+        public void WriteMetric(double value) { }
+    }
+    public partial class PollingCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public PollingCounter(string name, System.Diagnostics.Tracing.EventSource eventSource, Func<double> metricProvider) : base(name, eventSource) { }
+    }
+    public partial class IncrementingEventCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public IncrementingEventCounter(string name, System.Diagnostics.Tracing.EventSource eventSource) : base(name, eventSource) { }
+        public void Increment(double increment = 1) { }
+        public TimeSpan DisplayRateTimeScale { get { throw null; } set { } }
+    }
+    public partial class IncrementingPollingCounter : System.Diagnostics.Tracing.DiagnosticCounter
+    {
+        public IncrementingPollingCounter(string name, System.Diagnostics.Tracing.EventSource eventSource, Func<double> totalValueProvider) : base(name, eventSource) { }
+        public TimeSpan DisplayRateTimeScale { get { throw null; } set { } }
```

Comment:
What about tests?

Reply:
There are some tests that are checked in CoreCLR: https://github.com/dotnet/coreclr/tree/master/tests/src/tracing/eventcounter. I've tested these end-to-end scenarios using dotnet-counters (https://github.com/dotnet/diagnostics/tree/master/src/Tools/dotnet-counters), but I'll be checking in smaller tests to CoreCLR soon.